adr/0002-reactive-rename.adoc

```diff
@@ -0,0 +1,192 @@
+
+Examples of good and bad names are:
+
+[%header,format="csv"]
+|===
+Extension, Good/Bad
+quarkus-hibernate-reactive, Good
+quarkus-resteasy-reactive, Bad
+smallrye-reactive-messaging, Bad 
+reactive-mysql-client, Good
+|===
+
+For majority of the extensions we can find a good name that causes no reuse of name, but for the REST Client we have a problem since the name `quarkus-rest-client` is already taken by the legacy REST Client implemented in RESTEasy Classic. Thus the best option is if we could come up with a name base name for `quarkus-rest`, i.e. `quarkus-restless` or similar but thus far we have not been able to come up with a good name, that is better or as acceptable as `quarkus-rest`.
+
+Furthermore, we need to find a balance between the pain of renaming and the pain of keeping the status quo - while also informing/educating extension developers and users about the change in naming.
+
+== Considered options
+
+=== Do nothing
+
+We can continue with the current situation, but that feels less than ideal given the recurrence of the problem.
+
+=== Rename 
+
+We could simply rename the extensions without providing any backward compatibility with the old maven modules. This would likely cause a lot of pain - even in the presence of the Quarkus CLI update process, a lot of users won't use it.
+
+=== Rename to new name everywhere
+
+This would have the least impact on upgrading users, but it would also mean that we would have to come up with a new base-name for the REST Client. At this point no such name have been found.
+
+=== Partial rename to new new name
+
+For example, use `quarkus-rest` as base for everything, rename `quarkus-rest-client` to `quarkus-resteasy-client` and keep `quarkus-rest-client-reactive` as is. This would be a compromise between the two previous options. 
+
+Issue here is that it makes everything more complex and inconsistent.
+
+=== Rename + warn/break at build time
+
+Here we rename to new names everywhere but uses relocation's to warn/inform for dependency updates and add further logic in Quarkus buildtime phase to detect incompatible combinations. 
+
+For example today we detect when extensions are incompatible (i.e. resteasy combined resteasy-reactrive) with each other and fail the build. We could extend this to also detect when extensions are incompatible with each other due to the new naming.
+
+== Decision
+
+The decision is that we MUST do something about removing the word "reactive" from the names of the extensions that are not forcing a reactive programming model. 
+
+At a highlevel that means we will do the following:
+
+* Rename current `smallrye-messaging` to `quarkus-messaging`
+* Rename current `resteasy-reactive` to `quarkus-rest`
```

Comment:
I think you need to give it proper form, it's strange to use `resteasy-reactive` for `quarkus-resteasy-reactive` because if you decided to leave out `quarkus-` prefix, you also need to leave it out from `quarkus-rest`. Same goes for `rest-client` example below.

Reply:
You are not supposed to leave out the `quarkus-` prefix and `quarkus-rest` is a name we already use in several places. This is cleaning that up so its consistent. 

I'm going to mark this as resolved. if you still think there is an issue that is relevant for the intent here of this PR  add more context/info.

Reply:
I just wanted exactly what is now Clement suggesting, probably I didn't explain it right. Resolving the comment is fine, we have Clement comments. Thanks